lib/nerves_runtime/configfs.ex

```diff
@@ -0,0 +1,153 @@
+
+    Enum.each(Enum.reverse(mani.folders), fn folder ->
+      mkdir_p(Path.join("/sys/kernel/config/", folder))
+    end)
+
+    Enum.each(Enum.reverse(mani.files), fn {file, value} ->
+      write(Path.join("/sys/kernel/config/", file), value)
+    end)
+  end
+
+  defp apply_link(patha, pathb) do
+    ln_s(patha, pathb)
+  end
+
+  defp mkdir_p(dir) do
+    {_, 0} = Runtime.cmd("mkdir", ["-p", dir], :return)
+  end
+
+  defp ln_s(patha, pathb) do
+    {_, 0} = Runtime.cmd("ln", ["-s", patha, pathb], :return)
+  end
+
+  defp write(path, value) do
+    {_, 0} = Runtime.cmd("sh", ["-c", "echo #{value} > #{path}"], :return)
+  end
+
+  def build_manifest(map, state \\ %{path: "/", files: [], folders: []})
+
+  def build_manifest(%{} = map, state) do
+    build_manifest(Map.to_list(map), state)
+  end
+
+  def build_manifest([{key, %{} = val} | rest], state) when map_size(val) == 0 do
+    folder = Path.join(state.path, key)
+    build_manifest(rest, %{state | path: state.path, folders: (state.folders -- [folder]) ++ [folder]})
+  end
+
+  def build_manifest([{key, %{} = val} | rest], state) do
+    state = build_manifest(val, %{state | path: Path.join(state.path, key)})
+    build_manifest(rest, state)
+  end
+
+  def build_manifest([{key, edge_node} | rest], state) do
+    build_manifest(rest, %{
+      state
+      | folders: (state.folders -- [state.path]) ++ [state.path],
+        files: state.files ++ [{Path.join(state.path, key), edge_node}]
+    })
+  end
+
+  def build_manifest([], state), do: %{state | path: Path.dirname(state.path)}
+
+  def gadget_config do
+    %{
+      "usb_gadget" => %{
+        "g" => %{
+          "bcdUSB" => "0x0200",
+          "bcdDevice" => "0x3000",
+          "bDeviceClass" => "2",
+          "idVendor" => "0x1d6b",
+          "idProduct" => "0x0104",
+          "os_desc" => %{
+            "use" => "1",
+            "b_vendor_code" => "0xcd",
+            "qw_sign" => "MSFT100"
+          },
+          "functions" => %{
+            "ecm.usb0" => %{
+              "dev_addr" => "02:1e:58:8a:8f:42",
+              "host_addr" => "12:1e:58:8a:8f:42"
+            },
+            "acm.usb0" => %{},
+            "rndis.usb0" => %{
+              "dev_addr" => "22:1e:58:8a:8f:42",
+              "host_addr" => "32:1e:58:8a:8f:42",
+              "os_desc" => %{
+                "interface.rndis" => %{
+                  "compatible_id" => "RNDIS",
+                  "sub_compatible_id" => "5162001"
+                }
+              }
+            }
+          },
+          "strings" => %{
+            "0x409" => %{
+              "manufacturer" => Nerves.Runtime.KV.get_active("nerves_fw_author"),
+              "product" => Nerves.Runtime.KV.get_active("nerves_fw_product"),
+              # This is obviously not a good idea.
+              "serialnumber" => Nerves.Runtime.cmd("cat", ["/proc/cpuinfo"], :return) |> elem(0) |> String.split("Serial\t\t: ") |> List.last() |> String.trim()
```

Comment:
This won't work on many of our systems. Maybe that's why you said that it isn't a good idea. Calling `boardid` can get you a better serial number, but it's not perfect either.